src/vm/eventpipeeventinstance.cpp

```diff
@@ -53,7 +53,15 @@ EventPipeEventInstance::EventPipeEventInstance(
 
     m_pData = pData;
     m_dataLength = length;
-    QueryPerformanceCounter(&m_timeStamp);
+
+    // Declare the timeStamp as a local and require that it be 4-byte aligned.
+    // If the timeStamp is not 4-byte aligned, QueryPerformanceCounter will return E_NOACCESS on some versions of Windows.
+    // This codepath is more susceptible to having an unaligned m_timeStamp because EventPipeEventInstance
+    // is initialized via placement-new into the EventPipe circular buffer.
+    DECLSPEC_ALIGN(4) LARGE_INTEGER timeStamp;
+    QueryPerformanceCounter(&timeStamp);
+    m_timeStamp.QuadPart = timeStamp.QuadPart;
```

Comment:
If this whole type can be allocated via placement new on arbitrary odd addresses, you may have problems in this and other places too. It is valid for the compiler to use instruction that assumes that the address is aligned properly. 

You may want to either get the placement sufficiently aligned; or copy all data over using memcpy.

Reply:
Ok, I will investigate this.